src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -175,22 +183,27 @@ public JObject Header
         }
 
         /// <summary>
-        /// Gets the 'value' of the 'JWT ID' claim { jti, ''value' }.
+        /// Gets the 'value' of the 'jti' claim { jti, ''value' }.
         /// </summary>
-        /// <remarks>If the 'JWT ID' claim is not found, an empty string is returned.</remarks>
-        public override string Id => Payload.Value<string>(JwtRegisteredClaimNames.Jti) ?? String.Empty;
+        /// <remarks>If the 'jti' claim is not found, an empty string is returned.</remarks>
+        public override string Id => Payload?.Value<string>(JwtRegisteredClaimNames.Jti) ?? String.Empty;
+
+        /// <summary>
+        /// Gets the <see cref="JsonWebToken"/> associated with this instance.
+        /// </summary>
+        public JsonWebToken InnerToken { get; internal set; }
 
         /// <summary>
         /// Gets the 'value' of the 'iat' claim { iat, 'value' } converted to a <see cref="DateTime"/> assuming 'value' is seconds since UnixEpoch (UTC 1970-01-01T0:0:0Z).
         /// </summary>
-        /// <remarks>If the 'exp' claim is not found, then <see cref="DateTime.MinValue"/> is returned.</remarks>
-        public DateTime IssuedAt => GetDateTime(JwtRegisteredClaimNames.Iat);
+        /// <remarks>If the 'iat' claim is not found, then <see cref="DateTime.MinValue"/> is returned.</remarks>
+        public DateTime IssuedAt => Payload == null ? DateTime.MinValue : GetDateTime(JwtRegisteredClaimNames.Iat);
```

Comment:
>DateTime.MinValue [](start = 54, length = 17)

Nullable<T> seems like a more appropriate contract for optional claims like 'iat', 'exp', 'nbf', etc. Otherwise, this requires consumers of this library to know the internal details, and check for special cases like this.

Reply:
I agree, it would be a breaking change. Given that this api has just been released, it may be a good time to consider changing this. We may not have another opportunity.

Reply:
I agree that allowing these types to be nullable is more intuitive for a user of the library. However, the 'exp' (ValidTo) and 'nbf' (ValidFrom) claims are represented as DateTime values on the SecurityToken base class. As a result, it's not possible to make them nullable on the JsonWebToken class.